File: Easy/0001-two-sum/python/twosum1.py
class Solution:
    def twoSum(self, nums, target):
        for i in range(len(nums)):                                          # Iterate through each element in the array
            for j in range(i + 1, len(nums)):                               # For each element, check if there's another element that adds up to the target
                if nums[i] + nums[j] == target:                             # If the sum equals the target, return the indices 
                    return [i, j]

File: Easy/0001-two-sum/c/twosum1.c
#include <stdio.h>
#include <stdlib.h>

// Function to find two indices such that their values add up to the target
int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    int* result = (int*)malloc(2*sizeof(int));         //Allocate memory for the indices
    for (int i=0; i<numsSize; i++) {                   //Iterate through each element of the given array
        for (int j= i+1; j<numsSize; j++) {            //For each element, check if there is another element that adds up to the target -> O(n^2) time complexity -> must be done better
            if (nums[i] + nums[j] == target) {         //If the sum of nums[i] and nums[j] = target, store the indices
                result[0]=i;
                result[1]=j;
                *returnSize=2;                         //Set the return size to 2 (since we're returning two indices)
                return result;                         //Return the indices
            }
        }
    }
    *returnSize = 0;                                   //If no solution is found, set the return size to 0
    return NULL;
}
